test/unit.cpp

```diff
@@ -12740,7 +12742,8 @@ TEST_CASE("regression tests")
 
         CHECK(j2a.dump() == "23.42");
         //issue #230
-        //CHECK(j2b.dump() == "23.42");
+        CHECK(j2b.dump() == "2.342e01");
```

Comment:
I made  ```CHECK(j2b.dump() == "2.342e01");``` for 2 reasons.

First, to match the existing cases:
```
        CHECK(j3b.dump() == "1E04");
        CHECK(j3c.dump() == "1e04");
```
Second, and more subtly, we fundamentally don't have enough information within ```m_type.bits``` to reconstruct the original string (we would have to know the original exponent).  Because of this, only floating point numbers in scientific notation can make a successful round-trip (hence all of the round-trip regression tests only use scientific notation).